### ISSUE TITLE
Email Notification via OAuth2

### ISSUE THREAD

**Issue by bdin1** [Cacti/cacti#5739]
# Email Notification via OAuth2

Current I see the following email methods
PHPMail()
Sendmail
SMTP

I'm running Version 1.2.26
Maybe this has been asked before, but so many email providers are looking for stronger security to send emails, having access to OAuth would be nice. I do see in the /cacti/include/vendor/phpmailer/src has a OAuth.php method, but not sure how to configure it in Cacti. I do have access to microsoft graph api (codes and token)

**Comment by TheWitness** [comment]
Thanks for raising this.  We will look into it.

**Comment by TheWitness** [comment]
So, which example works better for you option1 or option2?

https://github.com/PHPMailer/PHPMailer/blob/master/examples/gmail_xoauth.phps

Lines 54++ in that file.

**Comment by bdin1** [comment]
I think option 1 looks better, I think it could be modified to allow for Office 365 and then a call to lookup a stored token that is refreshed every hour 

 

Thank you,

Eric C

**Comment by TheWitness** [comment]
@bdin1, 

Just looking at this.  It's painful as there are so many settings to preserve, but it's also pretty strait forward.  I'm on vacation this week, but will start to rough this in next week.

Larry

**Comment by TheWitness** [comment]
Have you been able to spend any time here yet @xmacan?  I'm going through issues this week since I won't have dev access.

**Comment by xmacan** [comment]
I'll do it in the next few days

**Comment by bdin1** [comment]
I've installed  1.3.0 - Dev f09d5b2a on a test system to see about using Oauth2, I have a process already gets a token. We also use Microsoft as our mail provider

**Comment by xmacan** [comment]
Please wait few days. I'm working on it but I was busy. It is PR https://github.com/Cacti/cacti/pull/6024

**Comment by xmacan** [comment]
@bdin1 If you have token, you can use code from this PR now. I have tested it with google and it is working. I have only one issue with get token directly from cacti

**Comment by bdin1** [comment]
I did see that link, just not a way to use a token. Maybe I am missing something on how to use the OAuth2 process

**Comment by TheWitness** [comment]
Once the pull request is merged into the development Branch then there should be full oath2 to support including renewing your tokens.  @xmacan and I are negotiating as to how to move forward to get this into the develop branch sooner than later.

**Comment by bdin1** [comment]
wonderful, thank you

**Comment by TheWitness** [comment]
@bdin1 Okay, the pull request is in.  We have some issues, but we will handle them under QA.  Please test and open issues as you find them.

I will close this as completed now.  Open new tickets like I said when you run across issues.
